### PR TITLE
Allow for empty containers in snabb set

### PR DESCRIPTION
**Pull request by tsyesika**
This fixes a problem brought up in issue #700. If you omitted some containers which were entirely optional it would produce an obtuse error. This is because some of the containers were represented as simple ctypes and when trying to create those when passing in "nil" (when omitted) could cause a casting error in the FFI. Since omitting optional values or blocks is how this is designed and expected to work this has been solved for checking for nil.

The following now succeeds:
```
$ sudo ./snabb config set lwaftr1 /softwire-config/internal-interface/ "ip ::1; mac 22:22:22:22:22:22; next-hop {}"
```

And the query that was performed in the issue itself now reports a more sensible error prompting for missing mandatory values:
```
$ sudo ./snabb config set lwaftr "/softwire-config/internal-interface/" "next-hop {}"
lib/yang/data.lua:266: missing scalar value: mac
stack traceback:
	core/main.lua:149: in function <core/main.lua:147>
	[C]: in function 'error'
	lib/yang/data.lua:266: in function 'finish'
	lib/yang/data.lua:412: in function 'parser'
	program/config/common.lua:78: in function 'parse_command_line'
	program/config/set/set.lua:7: in function 'run'
	program/config/config.lua:23: in function 'run'
	core/main.lua:67: in function <core/main.lua:43>
	[C]: in function 'xpcall'
	core/main.lua:211: in main chunk
	[C]: at 0x004545e0
	[C]: in function 'pcall'
	core/startup.lua:3: in main chunk
	[C]: in function 'require'
	[string "require "core.startup""]:1: in main chunk
```

### CODE REVIEW

File: src/lib/yang/data.lua [hunk 1]
```diff
@@ -204,9 +204,16 @@ local function struct_parser(keyword, members, ctype)
       return parse1(P)
    end
    local struct_t = ctype and typeof(ctype)
-   local function finish(out)
-      -- FIXME check mandatory values.
-      if struct_t then return struct_t(out) else return out end
+   local function finish(out, leaf)
+     -- FIXME check mandatory values.
```
Comment (teknico) [inline src/lib/yang/data.lua:208]: Is this comment still valid?

**Review by teknico — APPROVED** [review]
LGTM with one question.

File: src/lib/yang/data.lua (hunk 1 above)
Comment (tsyesika) [reply src/lib/yang/data.lua:208]: I believe, I've not added any checks for that. There are checks on simple scalar values which is what you see produce the error message in my example but no checks on struct values which still need to be added. This simply allows for omitting optional arguments. Previously it would error when you tried to do `struct_t(nil)` (when `out` was omitted and therefore nil)).

TLDR: yep, i think we still need it :)
